src/test/java/emissary/core/channels/BufferedChannelFactoryTest.java

```diff
@@ -0,0 +1,28 @@
+package emissary.core.channels;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class BufferedChannelFactoryTest {
+    @Test
+    void testCache() throws IOException {
+
+        final byte[] bytes = new byte[67];
+        final SeekableByteChannelFactory bytesSbcf = InMemoryChannelFactory.create(bytes);
+
+        Arrays.fill(bytes, (byte) 1);
+
+        assertThrows(NullPointerException.class, () -> BufferedChannelFactory.create(null, 10));
+        assertThrows(IllegalArgumentException.class, () -> BufferedChannelFactory.create(bytesSbcf, -1));
+
+        for (int blockSize = 1; blockSize < bytes.length; blockSize++) {
+            final SeekableByteChannelFactory bufferedSbcf = BufferedChannelFactory.create(bytesSbcf, blockSize);
+
+            ChannelTestHelper.checkByteArrayAgainstSbc(bytes, bufferedSbcf);
```

Comment:
This validation would probably be more useful if the test bytes weren't all `1`.

Reply:
Updated to fill the array elements with different values.